### PR TITLE
Unit Payload spawning units on block removing/destruction

### PR DESCRIPTION
**Pull request by Slava0135**
Peek 2020-11-28 20-38

also minor typo fix in comments

### CODE REVIEW

File: core/src/mindustry/world/blocks/distribution/PayloadConveyor.java [hunk 1]
```diff
@@ -234,6 +234,18 @@ public void handlePayload(Building source, Payload payload){
             updatePayload();
         }
 
+        @Override
+        public void onDestroyed(){
+            super.onDestroyed();
+            if (item != null) item.dump();
```
Comment (Anuken) [inline core/src/mindustry/world/blocks/distribution/PayloadConveyor.java:240]: formatting incorrect: no space after the if

Comment (Quezler) [inline core/src/mindustry/world/blocks/distribution/PayloadConveyor.java:240]:
```suggestion
            if(item != null) item.dump();
```

File: core/src/mindustry/world/blocks/distribution/PayloadConveyor.java [hunk 2]
```diff
@@ -234,6 +234,18 @@ public void handlePayload(Building source, Payload payload){
...
+        }
+
+        @Override
+        public void onRemoved(){
+            super.onRemoved();
+            if (item != null) item.dump();
```
Comment (Quezler) [inline core/src/mindustry/world/blocks/distribution/PayloadConveyor.java:246]:
```suggestion
            if(item != null) item.dump();
```

File: core/src/mindustry/world/blocks/production/PayloadAcceptor.java [hunk 3]
```diff
@@ -82,6 +82,18 @@ public Payload takePayload(){
             return t;
         }
 
+        @Override
+        public void onDestroyed(){
+            super.onDestroyed();
+            if (payload != null) payload.dump();
```
Comment (Quezler) [inline core/src/mindustry/world/blocks/production/PayloadAcceptor.java:88]:
```suggestion
            if(payload != null) payload.dump();
```

File: core/src/mindustry/world/blocks/production/PayloadAcceptor.java [hunk 4]
```diff
@@ -82,6 +82,18 @@ public Payload takePayload(){
...
+        }
+
+        @Override
+        public void onRemoved(){
+            super.onRemoved();
+            if (payload != null) payload.dump();
```
Comment (Quezler) [inline core/src/mindustry/world/blocks/production/PayloadAcceptor.java:94]:
```suggestion
            if(payload != null) payload.dump();
```

**Review by Quezler — CHANGES_REQUESTED** [review]
owo

**Pull request merged**